### PR TITLE
Fix Undo/Redo, Bump Dependencies

### LINKED ISSUE #201: 🐞 Using ⌘ + Z will crash CodeEdit
### Description

It is possible, but not so common, for CodeEdit to crash when using ⌘ + Z, this seems to be part of a general issue with the Undo feature

### To Reproduce

1. Open any file
2. Go to the end of the file and press Enter + Backspace at least two times
3. Use ⌘ + Z multiple times until the application crashes

### Expected Behavior

The previous actions should be restored

### Version Information

CodeEdit: 0.0.3-alpha

### Additional Context

_No response_

### Screenshots

https://github.com/CodeEditApp/CodeEdit/assets/62521219/69bfd1db-5c13-49bc-830a-d78238521195

### LINKED ISSUE #203: 🐞 Unexpected Undo behavior
### Description

When you edit text and undo changes you made, the undo(s) would be realized at some point but doesn't recover previous state and has some glitches in the text you didn't edit.

### To Reproduce

1. Open a file (`.swift` file in this case, and I realized the app doesn't support opening `.xcodeproj` file, which is not the point here though).
2. Edit few lines like:
Screenshot 2023-04-15 at 7 14 31 PM
3. Perform `Cmd+Z` multiple times
4. You get something like this:
Screenshot 2023-04-15 at 7 14 52 PM

### Expected Behavior

Bring back the previous state as it was originally.

### Version Information

CodeEdit: 0.0.3-alpha-32
macOS: 13.3.1 (MBP "14 2021)
Xcode: 14.3 *(I did install the app **from the released installer, not from source**)*

### PR DESCRIPTION
**Pull request by thecoolwinter**
### Description

Bumps both `STTextView` and `TextFormation` to their latest versions. This version of `TextFormation` changes how whitespaces are handled by filters, so there were some small modifications that needed to be made to accommodate. This also slightly adjusts the `DeleteWhitespaceFilter` to match other editor's functionality. 

`DeleteWhitespaceFilter` now deletes only the leading whitespace, and only to the nearest column, instead of just a number of spaces. It will also jump to the rightmost side of the whitespace when deleting.

The other half of this PR is an implementation for undo/redo. This adds a `CEUndoManager` (with the prefix b/c AppKit declares that type) class that implements an undo/redo stack and operation grouping. It provides methods for performing undo/redo operations, registering text mutations from the editor, grouping operations, and clearing the undo stack. The class also automatically groups certain mutations. See the class documentation for more info there.

### Related Issues

* Closes #201 
* Closes #203 

### Checklist

- [x] I read and understood the [contributing guide](https://github.com/CodeEditApp/CodeEdit/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/CodeEditApp/CodeEdit/blob/main/CODE_OF_CONDUCT.md)
- [x] The issues this PR addresses are related to each other
- [x] My changes generate no new warnings
- [x] My code builds and runs on my machine
- [x] My changes are all related to the related issue above
- [x] I documented my code

### Screenshots

https://github.com/CodeEditApp/CodeEditTextView/assets/35942988/35c64ed6-590d-493a-8f91-d918fef363a5

### CODE REVIEW

**Review by bombardier200 — APPROVED** [review]
Some great progress made here!

**Review by 0xWDG — APPROVED** [review]
Nice work!

File: Sources/CodeEditTextView/Extensions/NSTextLayoutManager+textLineFragment.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//
+//  File.swift
+//  
+//
+//  Created by Khan Winter on 7/8/23.
+//
+
+import AppKit
+
+extension NSTextLayoutManager {
```
Comment (krzyzanowskim) [inline Sources/CodeEditTextView/Extensions/NSTextLayoutManager+textLineFragment.swift:10]: It looks like copy from `STTextView`.

FYI this is exposed as a public API in [STTextKitPlus](https://github.com/krzyzanowskim/STTextView/blob/d62b3a2ee9bde334f74c94209f287f4acc33d60d/Package.swift#L13) module now.

Comment (thecoolwinter) [reply Sources/CodeEditTextView/Extensions/NSTextLayoutManager+textLineFragment.swift:10]: Tbh it was, we relied on the extension. Thanks for the heads up!

**Pull request merged**